File: 12-06-2025_Maximum Difference Between Adjacent Elements in a Circular Array.cpp
class Solution {
public:
    int maxAdjacentDistance(vector<int>& nums) {
        int maxdiff=abs(nums[nums.size()-1]-nums[0]);

        if (nums.size() < 2) return 0; 
        for(int i=0;i<nums.size()-1;i++)
        {
            if(maxdiff<abs(nums[i]-nums[i+1]))
            {
                maxdiff=abs(nums[i]-nums[i+1]);
            }
        }
        return maxdiff;
    }
};

File: 12-06-2025_Maximum Difference Between Adjacent Elements in a Circular Array.py

class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        if len(nums) < 2:
            return 0
        
        max_diff = abs(nums[-1] - nums[0])
        
        for i in range(len(nums) - 1):
            current_diff = abs(nums[i] - nums[i + 1])
            if current_diff > max_diff:
                max_diff = current_diff
                
        return max_diff